e2e/execution_test.go

```diff
@@ -89,6 +89,87 @@ func testExecution(t *testing.T) {
+			inputs         = &types.Struct{
+				Fields: map[string]*types.Value{
+					"msg": {
+						Kind: &types.Value_StringValue{
+							StringValue: "test",
+						},
+					},
+				},
+			}
+		)
+		t.Run("create first", func(t *testing.T) {
+			resp, err := client.ExecutionClient.Create(context.Background(), &pb.CreateExecutionRequest{
+				TaskKey:      taskKey,
+				EventHash:    hash.Int(2),
+				ExecutorHash: executorHash,
+				Inputs:       inputs,
+			})
+			require.NoError(t, err)
+			executionHash1 = resp.Hash
+		})
+		t.Run("create second", func(t *testing.T) {
+			resp, err := client.ExecutionClient.Create(context.Background(), &pb.CreateExecutionRequest{
+				TaskKey:      taskKey,
+				EventHash:    hash.Int(3),
+				ExecutorHash: executorHash,
+				Inputs:       inputs,
+			})
+			require.NoError(t, err)
+			executionHash2 = resp.Hash
+		})
+		t.Run("first in progress", func(t *testing.T) {
+			execInProgress, err := stream.Recv()
+			require.NoError(t, err)
+			require.Equal(t, executionHash1, execInProgress.Hash)
+			require.Equal(t, taskKey, execInProgress.TaskKey)
+			require.Equal(t, hash.Int(2), execInProgress.EventHash)
+			require.Equal(t, executorHash, execInProgress.ExecutorHash)
+			require.Equal(t, execution.Status_InProgress, execInProgress.Status)
+			require.True(t, inputs.Equal(execInProgress.Inputs))
+		})
+		t.Run("second in progress", func(t *testing.T) {
+			execInProgress, err := stream.Recv()
+			require.NoError(t, err)
+			require.Equal(t, executionHash2, execInProgress.Hash)
+			require.Equal(t, taskKey, execInProgress.TaskKey)
+			require.Equal(t, hash.Int(3), execInProgress.EventHash)
+			require.Equal(t, executorHash, execInProgress.ExecutorHash)
+			require.Equal(t, execution.Status_InProgress, execInProgress.Status)
+			require.True(t, inputs.Equal(execInProgress.Inputs))
+		})
+		t.Run("first completed", func(t *testing.T) {
+			exec, err := stream.Recv()
+			require.NoError(t, err)
+			require.Equal(t, executionHash1, exec.Hash)
+			require.Equal(t, taskKey, exec.TaskKey)
+			require.Equal(t, hash.Int(2), exec.EventHash)
+			require.Equal(t, executorHash, exec.ExecutorHash)
+			require.Equal(t, execution.Status_Completed, exec.Status)
+			require.True(t, inputs.Equal(exec.Inputs))
+			require.Equal(t, "test", exec.Outputs.Fields["msg"].GetStringValue())
+			require.NotEmpty(t, exec.Outputs.Fields["timestamp"].GetNumberValue())
+		})
+		t.Run("second completed", func(t *testing.T) {
+			exec, err := stream.Recv()
+			require.NoError(t, err)
+			require.Equal(t, executionHash2, exec.Hash)
+			require.Equal(t, taskKey, exec.TaskKey)
+			require.Equal(t, hash.Int(3), exec.EventHash)
+			require.Equal(t, executorHash, exec.ExecutorHash)
+			require.Equal(t, execution.Status_Completed, exec.Status)
+			require.True(t, inputs.Equal(exec.Inputs))
+			require.Equal(t, "test", exec.Outputs.Fields["msg"].GetStringValue())
+			require.NotEmpty(t, exec.Outputs.Fields["timestamp"].GetNumberValue())
```

Comment:
I think we don't need to check so many thing there. This test is "over-asserted" and should only compare if two executions were completed. None of  `in progress` tests or any tests for hashes are needed.

Reply:
Is it really the assertion that you don't like or the number of lines it uses? I was thinking to create assertion function for executions.

Reply:
> Is it really the assertion that you don't like or the number of lines it uses? 

Both. The number of lines is tremendous here and many of the assertions are not needed in that test. For me, the test should focus on what is actually tested - nothing more, nothing less.

So:

1) task1 inputs are optional so we don't need to create it here, we can leave it empty
2) receiving in-progress is not needed - we can only check for a completed task
3) assertion of every field is not needed - execution hash probably would be enough (or use Equal function from protobuf as you mentioned)